cfme/utils/appliance/__init__.py

```diff
@@ -1510,6 +1512,38 @@ def wait_for_web_ui(self, timeout=900, running=True, log_callback=None):
             fail_condition=not running, delay=10)
         return result
 
+    def wait_for_api_available(self, num_sec=600):
+        """Waits for the web UI to be running / to not be running
+
+        Args:
+            num_sec: Number of seconds to wait until num_sec(default ``600``)
+        """
+
+        def _check_appliance_api_ready():
+            try:
+                # There are 2 hard problems in computer science: cache
+                # invalidation, naming things, and off-by-1 errors.
+                # -- Leon Bambrick
```

Comment:
:smile:

Reply:
Hilarious, and great implementation of the wait/check pattern here @jaryn !